cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/DefaultSecureStoreService.java

```diff
@@ -61,11 +64,13 @@
   }
 
   /**
-   * Lists all the secure keys in the given namespace that the user has access to.
+   * Lists all the secure keys in the given namespace that the user has access to. Returns an empty list if the user
+   * does not have access to the namespace or any of the keys in the namespace.
    * @param namespaceId Id of the namespace we want the key list for.
-   * @return A list of {@link SecureKeyListEntry} of all the keys visible to the user under the given namespace.
-   * @throws Exception If there was a problem getting the list from the underlying provider.
-   * or if we fail to create a filter for the current principal.
+   * @return A list of {@link SecureKeyListEntry} for all the keys visible to the user under the given namespace.
+   * @throws NamespaceNotFoundException If the specified namespace does not exist.
+   * @throws IOException If there was a problem reading from the store.
+   * @throws Exception If we fail to create a filter for the current principal.
```

Comment:
I think we should just remove this, and instead add an @throws for UnauthorizedException